daemon/cmd/daemon_main.go

```diff
@@ -1714,16 +1722,44 @@ func newDaemonPromise(params daemonParams) (promise.Promise[*Daemon], promise.Pr
 			daemon = d
 
 			if !option.Config.DryMode {
-				if err := startDaemon(daemon, restoredEndpoints, cleaner, params); err != nil {
-					daemonResolver.Reject(err)
-					cancelDaemonCtx()
-					cleaner.Clean()
-					cfgResolver.Reject(err)
-					return err
+				log.Info("Initializing daemon")
+
+				// This validation needs to be done outside of the agent until
+				// datapath.NodeAddressing is used consistently across the code base.
+				log.Info("Validating configured node address ranges")
+				if err := node.ValidatePostInit(); err != nil {
+					return fmt.Errorf("postinit failed: %w", err)
+				}
+
+				// Store config in file before resolving the DaemonConfig promise.
+				err = option.Config.StoreInFile(option.Config.StateDir)
+				if err != nil {
+					log.WithError(err).Error("Unable to store Cilium's configuration")
+				}
+
+				err = option.StoreViperInFile(option.Config.StateDir)
+				if err != nil {
+					log.WithError(err).Error("Unable to store Viper's configuration")
 				}
 			}
-			daemonResolver.Resolve(daemon)
+
+			// NOTE: 'option.Config' must not be modified after this point
```

Comment:
The `option.Config` currently can be modified during the entire runtime take `option.Config.Opts` for example. So this isn't entirely accurate.

Reply:
Done.

Reply:
That was the only one I know for sure. Lets assert the rest is stable and see what errors shake loose